src/chordsymbol.ts

```diff
@@ -395,227 +222,123 @@ export class ChordSymbol extends Modifier {
     return true;
   }
 
-  /** Currently unused. */
-  protected static noFormat: boolean = false;
-
   protected symbolBlocks: ChordSymbolBlock[] = [];
   protected horizontal: number = ChordSymbolHorizontalJustify.LEFT;
   protected vertical: number = ChordSymbolVerticalJustify.TOP;
-  protected useKerning: boolean = true;
-  protected reportWidth: boolean = true;
-
-  // Initialized by the constructor via this.setFont().
-  protected textFormatter!: TextFormatter;
 
   constructor() {
     super();
-    if (Tables.currentMusicFont().getName() === 'Petaluma') {
-      // Fixes Issue #1180
-      this.textFont.family = 'PetalumaScript, Arial, sans-serif';
-    }
-
-    this.textFormatter = TextFormatter.create(this.textFont);
   }
 
   /**
    * The offset is specified in `em`. Scale this value by the font size in pixels.
    */
   get superscriptOffset(): number {
-    return ChordSymbol.superscriptOffset * this.textFormatter.fontSizeInPixels;
+    return ChordSymbol.superscriptOffset * Font.convertSizeToPixelValue(this.textFont?.size);
```

Comment:
I see you do this twice:
```
this.textFont?.size
```

Can textFont be undefined? Why are we doing the optional chaining `?.` operator here?

If it is possible for it to be undefined, it will pass undefined to `Font.convertSizeToPixelValue(...)`. I don't believe that function can accept an undefined parameter.

Can we remove the ? from the two places we use it? I thought textFont is required to be defined.

Reply:
done